scripts/tools/generate_esp32_chip_factory_bin.py

```diff
@@ -277,40 +277,48 @@ def gen_spake2p_params(passcode):
 
 
 def populate_factory_data(args, spake2p_params):
-    FACTORY_DATA['discriminator']['value'] = args.discriminator
-    FACTORY_DATA['iteration-count']['value'] = spake2p_params['Iteration Count']
-    FACTORY_DATA['salt']['value'] = spake2p_params['Salt']
-    FACTORY_DATA['verifier']['value'] = spake2p_params['Verifier']
-    FACTORY_DATA['dac-cert']['value'] = os.path.abspath(args.dac_cert)
-    FACTORY_DATA['pai-cert']['value'] = os.path.abspath(args.pai_cert)
-    FACTORY_DATA['cert-dclrn']['value'] = os.path.abspath(args.cd)
-    FACTORY_DATA['dac-key']['value'] = os.path.abspath('dac_raw_privkey.bin')
-    FACTORY_DATA['dac-pub-key']['value'] = os.path.abspath('dac_raw_pubkey.bin')
-
-    if args.serial_num is not None:
+    if args.discriminator:
+        FACTORY_DATA['discriminator']['value'] = args.discriminator
+
+    if spake2p_params:
+        FACTORY_DATA['iteration-count']['value'] = spake2p_params['Iteration Count']
+        FACTORY_DATA['salt']['value'] = spake2p_params['Salt']
+        FACTORY_DATA['verifier']['value'] = spake2p_params['Verifier']
+
+    if args.dac_cert:
+        FACTORY_DATA['dac-cert']['value'] = os.path.abspath(args.dac_cert)
+    if args.pai_cert:
+        FACTORY_DATA['pai-cert']['value'] = os.path.abspath(args.pai_cert)
+    if args.cd:
+        FACTORY_DATA['cert-dclrn']['value'] = os.path.abspath(args.cd)
+    if args.dac_key:
+        FACTORY_DATA['dac-key']['value'] = os.path.abspath('dac_raw_privkey.bin')
+        FACTORY_DATA['dac-pub-key']['value'] = os.path.abspath('dac_raw_pubkey.bin')
+
+    if args.serial_num:
```

Comment:
same question here: any of these are numbers? line id_uid, serial_number, vendor_id etc? If any of them are acceptable to be 0, then the check should stay for 'None'

Reply:
Only checking numbers against `None` and others are left as is. discriminator, passcode, vendor id, product id, hardware version are numbers and other are string or boolean.